guides-markdown/wallets.md

```diff
@@ -243,12 +243,26 @@ console.log(jdWalletKey0.getPrivateKey('base58'));
 //cNZfR3NhQ9oCP3pTjvPZETUuTWZo2k6EXtfczvbWyv7FdjMhppvJ
 ```
 
+#### Using an HD path
+
+Sometimes it is handy to work with an HD path. This is an object that contains information on a particular path to a key in the tree hierarchy of the wallet. In particular, a path knows the account, the branch and the index of the relevant key. This is according to [BIP44](https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki), which specifies that every account gets its own subtree of keys. According to BIP44, an account has two children (known as branches): the first for receiveing addresses and the second for change addresses. Here in bcoin, there is also a third branch, for nested segwit addresses. (bcoin does not implement BIPS [49](https://github.com/bitcoin/bips/blob/master/bip-0049.mediawiki) and [84](https://github.com/bitcoin/bips/blob/master/bip-0084.mediawiki) yet.) These three branches can in turn have as many children as needed. These correspond to the aforementioned index and provide the actual keys that are used for transactions.
+
+```javascript
+const path = await wdb.getPath(wallet.wid, jdAccount.receiveKey().getHash())
+// Let's skip 10 indexes
+path.index += 10
+
+// Get the corresponding keyring:
+const jdKey = jdAccount.accountKey.derive(path.branch).derive(path.index)
+const keyring = WalletKey.fromHD(jdAccount, jdKey, path.branch, path.index)
+```
+
```

Comment:
Explanation is good, thank you! Tested the code examples and it works - last thing, lets just `console.log` the `keyring` here (with example output included in a comment, like the above sections) so `index: 10` is visible, then we'll merge. Only other thing to discuss, should we include a warning about generating addresses that are not "in order" from the wallet? The BIP44 spec has a "look ahead" value (20, but in bcoin its 10) but if you receive money at an address beyond that, its possible it will not be discovered if you recover your wallet from the mnemonic phrase. Might be worth mentioning.

Reply:
Good point. I suggest we increase `index` by 30 in the example and change the comment to something like `// if this keyring is used for receiving coins, jd's wallet will realize it only after  20 more receiving addresses are generated in the usual manner, because bcoin wallets look for received coins 10 addresses ahead of the last generated. Surprise money!`

Reply:
Yeah... but phrase it more like a warning. I don't want users sending money to index 100 and then losing it.